libraries/AP_Scheduler/AP_Scheduler.cpp

```diff
@@ -389,6 +400,50 @@ void AP_Scheduler::Log_Write_Performance()
     AP::logger().WriteCriticalBlock(&pkt, sizeof(pkt));
 }
 
+// display task statistics as text buffer for @SYS/tasks.txt
+size_t AP_Scheduler::task_info(char *buf, size_t bufsize) const
+{
+    size_t total = 0;
+
+    // a header to allow for machine parsers to determine format
+    int n = snprintf(buf, bufsize, "TasksV1\n");
```

Comment:
I think the hal.util->snprintf method is good.

Reply:
Yes true, although threads.txt does it this way - I seem to be getting asked a lot these days to fix code I copied from @tridge  ;)

Reply:
Fixed